### ISSUE TITLE
add support for response compression

### ISSUE THREAD

**Issue by tanner0101** [vapor/http#280]
Moving https://github.com/vapor/vapor/issues/1565 here

**Comment by bre7** [comment]
Implemented in https://github.com/apple/swift-nio/blob/master/Sources/NIOHTTP1/HTTPResponseCompressor.swift#L59 ?

**Comment by tanner0101** [comment]
Yup, just need to add support for enabling that pipeline handler here.

**Comment by marekpridal** [comment]
There is a new parameter in `HTTPServer.start` method but it's missing in NIOServerConfig so unable to set it in `configure` method.
